Documentation/AdvancedBuildInstructions.md

```diff
@@ -224,10 +224,7 @@ clangd that are aware of SerenityOS as a valid target. These tools will be insta
 the script. Pointing your editor's plugins to the custom-built clang tools and a ``compile_commands.json`` from a clang build
 of Serenity can enable richer error reporting than the tools that are installed for the build host.
 
-To enable building clangd as part of the clang toolchain, set ``CLANG_ENABLE_CLANGD`` to ON in
-``Toolchain/CMake/LLVMConfig.cmake`` before running BuildClang.sh. If you already built the clang toolchain and would like to
```

Comment:
This seems like a reasonable QOL change. But, the text for "if you already built the clang toolchain" and beyond should be kept. The LLVMConfig.cmake settings are only applied to the initial cache of the llvm build.

Reply:
As you may see I am forcefully setting the value, it doesn't matter whether you have already built the toolchain or not, regardless to that it will enable clangd and build/rebuild the targets.

Maybe I am not getting why you want to keep it. Care to explain?

Reply:
We pass LLVMConfig.cmake to cmake via `-C` in BuildClang.sh. From the cmake docs:

https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-C

> -C <initial-cache>
>
>    Pre-load a script to populate the cache.
>
>    When CMake is first run in an empty build tree, it creates a CMakeCache.txt file and populates it with customizable settings for the project. This option may be used to specify a file from which to load cache entries before the first pass through the project's CMake listfiles. The loaded entries take priority over the project's default values. The given file should be a CMake script containing [set()](https://cmake.org/cmake/help/latest/command/set.html#command:set) commands that use the CACHE option, not a cache-format file.

This option only affects the *initial* cache, and when run on an empty build tree. If you want to build clangd after you have created an llvm build directory, you need to use the steps I had previously documented there. Your new environment variable only has an effect when you don't have a clang build already. You can't re-run BuildClang.sh and have it take effect, because the build dir already exists. you would need to rm -rf your build directory, which will just end up doing another full build, instead of just adding clangd..

Reply:
Yes, that would overwrite the cache. But as I mentioned, by using `FORCE` option it is working for me.

```shell
rm -rf Toolchain/{Build,Local,Tarballs}
Toolchain/BuildClang.sh

# This will fail
file Toolchain/Local/clang/bin/clangd

CLANG_ENABLE_CLANGD=ON Toolchain/BuildClang.sh
# This will work
file Toolchain/Local/clang/bin/clangd
```

This is a list of build target while executing `CLANG_ENABLE_CLANGD=ON ./BuildClang.sh` command.

![image](https://github.com/SerenityOS/serenity/assets/28386721/d119142d-e1a7-40a6-a77c-df1ce5f2ac00)

Reply:
@ADKaster Ping 0x00